packages/chat-stateful-client/src/ChatClientState.ts

```diff
@@ -45,3 +51,82 @@ export type ChatThreadClientState = {
 export type ChatThreadProperties = {
   topic?: string;
 };
+
+/**
+ * Errors teed from API calls to the Chat SDK.
+ *
+ * Each property in the object stores the latest error for a particular SDK API method.
+ *
+ * Errors from this object can be cleared by calling the TODO(implement me) {@Link clearError} method.
+ * Additionally, errors are automatically cleared when:
+ * - The state is cleared.
+ * - Subsequent calls to related API methods succeed.
+ *   See documentation of individual stateful client methods for details on when errors may be automatically cleared.
+ */
+export type ChatErrors = {
+  [target in ChatErrorTargets]: Error;
+};
+
+/**
+ * Error thrown from failed stateful API methods.
+ */
+export class ChatError extends Error {
+  /**
+   * The API method target that failed.
+   */
+  public target: ChatErrorTargets;
+  /**
+   * Error thrown by the failed SDK method.
+   */
+  public inner: Error;
+
+  constructor(target: ChatErrorTargets, inner: Error) {
+    super();
+    this.target = target;
+    this.inner = inner;
+    this.name = 'ChatError';
+    this.message = `${this.target}: ${this.inner.message}`;
```

Comment:
Is this purely informational or do we expect developers to write conditionals of this?

Reply:
Actually better question, which property do we expect developers to write conditionals over?

Reply:
We expect them to use `this.target` and `this.inner`.

That's why I created a custom `Error` subclass over simply throwing a new `Error('${target}: ${inner}')` in the helper function.